docs/Reporting.md

```diff
@@ -1,58 +1,66 @@
-    2.  Run the command `yarn marge filename.json -f report -o /destination_path`
-    3.  Check the `html report` on the destination path.
-```
+## Using the Existing Reporting Logic
 
 ### Background
 
-TODO: add details
+The default reporting mechanism in FCA is designed to streamline your test analysis process. Post-execution of sanity tests, the app generates detailed mochawesome JSON reports. These reports are crucial for reviewing test outcomes. The `northBoundSchemaValidationAndReportGeneration()` function in [Test_Runner.js](../src/Test_Runner.js) orchestrates this process. Additionally, for users leveraging `IntentReader`, the [runTestHandler](../src/pubsub/handlers/RunTestHandler.js) class can be utilized to execute a sanity test and obtain a reports.
+
+#### Standalone Mode
+
+Furthermore, FCA supports `standalone` report generation. Standalone reporting posts to a different endpoint which allows for more diverse report management and usage.
+
+In order to activate `standalone` mode, include the following paramaters:
+
+- `standalone` (string: required): Set to `true` to enable standalone mode.
+
+- `reportingId` (string: optional): string - An identifier for naming the generated report JSON file.
+
+Parameters can be provided via intent or as URL query parameters. This mode is particularly useful for specialized report handling or when integrating with unique endpoint requirements.
 
 ### Setup
 
-TODO: add details
+To initiate the built-in reporting, follow the preliminary steps below.
 
-### Usage
+**NOTES**:
 
-TODO: add details
+- The steps below are tailored for AWS configurations. For other platforms, adjustments will be necessary.
 
-## Supported Report Parameters
+- `pushReportToS3()` is a veratile function that can interface with non-AWS POST endpoints to transfer your mochawesome JSON.
+
+1. Create a POST endpoint in AWS API Gateway to accept incoming JSON.
+
+2. Configure your endpoints in [config.js](../plugins/config.js) as follows, to set up the report publishing process:
 
-EXISTING CONTENTS(needs revision):
 ```
-- Schema: Schema validation response of each API’s. whether it passed/failed. Validation is done based on the Open RPC document
-- Content: Behavioural validation can be done.
-- Message: Defines the Schema of the API from the Open RPC document.
-- Actual: The API response which is invoked by FCA on the device is stored in actual.
-- Error: Based on the schema validation done by FCA, if the schema validation fails then reason of failure is stored in error.
+REPORT_PUBLISH_URL: '', // Endpoint for standard report publishing
+REPORT_PUBLISH_STANDALONE_URL: '', // Endpoint for standalone report publishing
+REPORT_PUBLISH_STANDALONE_REPORT_URL: '', // Optional: Endpoint for hosting a live HTML report
 ```
 
+### Usage
+
+Initiate a sanity test run to activate the reporting function. Upon completion, FCA automatically processes and generates the mochawesome JSON report.
+
+## Creating Your Own Reporting Logic
```

Comment:
Wouldn't the whatever you want come more in the endpoints you configured FCA to send reports to though? Not sure if we should be advising users to change that part of the code in FCA.

Reply:
I just feel like I'm missing the use case that requires this information.

Reply:
Yes agreed. Going to re-write this section and instead of encouraging users to make code changes, more so point explain that once the json gets posted to the endpoint they can do whatever they want with it.

Reply:
Re-wrote section and renamed to Extending Report Processing